src/main/java/org/apache/maven/plugins/dependency/tree/VerboseGraphSerializer.java

```diff
@@ -42,30 +42,370 @@
+        Queue<DependencyNode> queue = new LinkedList<>();
+        queue.add( root );
+        Artifact rootArtifact = root.getArtifact();
+
+        StringBuilder result = new StringBuilder( "digraph" );
+        result.append( " \"" ).append( rootArtifact.getGroupId() ).append( ":" ).append(
+                rootArtifact.getArtifactId() ).append( ":" ).append( rootArtifact.getExtension() ).append( ":" ).append(
+                rootArtifact.getVersion() ).append( "\" {" ).append( System.lineSeparator() );
+
+        while ( !queue.isEmpty() )
+        {
+            DependencyNode node = queue.poll();
+            for ( DependencyNode child : node.getChildren() )
+            {
+                result.append( " \"" );
+                String coordString = "";
+                boolean messageAdded = false;
+
+                if ( child.getArtifact().getProperties().containsKey( PRE_MANAGED_VERSION ) )
+                {
+                    coordString = coordString.concat(
+                            " - version managed from " + child.getArtifact().getProperties().get(
+                                    PRE_MANAGED_VERSION ) );
+                    messageAdded = true;
+                }
+                if ( child.getArtifact().getProperties().containsKey( PRE_MANAGED_SCOPE ) )
+                {
+                    if ( messageAdded )
+                    {
+                        coordString = coordString.concat( "; " );
+                    }
+                    else
+                    {
+                        coordString = coordString.concat( " - " );
+                        messageAdded = true;
+                    }
+                    coordString = coordString.concat(
+                            "scope managed from " + child.getArtifact().getProperties().get( PRE_MANAGED_SCOPE ) );
+                }
+                coordString = getDependencyCoordinate( child ) + coordString;
+                result.append( getDependencyCoordinate( node ) ).append( "\"" ).append( " -> \"" ).append(
+                        coordString );
+                if ( child.getArtifact().getProperties().containsKey( "Cycle" ) )
+                {
+                    if ( !messageAdded )
+                    {
+                        result.append( " -" );
+                    }
+                    result.append( " omitted due to cycle" );
+                }
+                else if ( nodeErrors.get( child ) != null )
+                {
+                    if ( !messageAdded )
+                    {
+                        result.append( " - " );
+                    }
+                    result.append( nodeErrors.get( child ) );
+                }
+                else if ( !visitedNodes.contains( child ) )
+                {
+                    visitedNodes.add( child );
+                    queue.add( child );
+                }
+                result.append( "\" ;" ).append( System.lineSeparator() );
```

Comment:
I'm not familiar with the DOT format. What line separator does it want? use that, not a platform dependent separator.